### ISSUE TITLE
HTTP server fails with "Address already in use" with dynamic ssl port

### ISSUE THREAD

**Issue by harthorst** [quarkusio/quarkus#9889]
**Describe the bug**
If I run quarkus with both http and https port=0 sometimes the server does not start. 

**Expected behavior**
Server starts using free ports

**Actual behavior**
Quarkus exits with: java.net.BindException: Address already in use

**To Reproduce**
1. git clone https://github.com/quarkusio/quarkus-quickstarts.git
2. cd quarkus-quickstarts/openapi-swaggerui-quickstart
3. mvn clean install
4. java -Dquarkus.http.ssl.certificate.file=localhost.crt -Dquarkus.http.ssl.certificate.key-file=localhost.key -Dquarkus.http.insecure-requests=redirect -Dquarkus.http.port=0 -Dquarkus.http.ssl-port=0 -jar target/openapi-swaggerui-quickstart-1.0-SNAPSHOT-runner.jar
5. repeat step 4 20 times :)

You may use server.pem and server.key file in grpc-tls-quickstart/src/main/resources/tls/ as certificate and key file.

**Configuration**
```
default
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin xxx 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
 - Output of `java -version`: 
```
java version "11.0.7" 2020-04-14 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.7+8-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.7+8-LTS, mixed mode)
```
 - Quarkus version or git rev: master
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /Users/thf/bin/apache-maven-3.6.3
Java version: 11.0.7, vendor: Oracle Corporation, runtime: /Users/thf/bin/jdk-11.0.7.jdk/Contents/Home
Default locale: de_DE, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.4", arch: "x86_64", family: "mac"
```

This behavior has been observed on windows and linux as well.

cc @famod

**Comment by famod** [comment]
cc @tkalmar

**Comment by jaikiran** [comment]
Hello @harthorst, is there a complete exception stacktrace that you could paste in this issue? That way it will be quicker to understand what's going on.

**Comment by harthorst** [comment]
[stacktrace.txt](https://github.com/quarkusio/quarkus/files/4762949/stacktrace.txt)
@jaikiran : sure. you'll find the complete log in the attachment.

**Comment by jaikiran** [comment]
For quick reference, here's the relevant part from that attachment:

```
2020-06-09 17:48:31,730 ERROR [io.qua.application] (main) Failed to start application: java.lang.RuntimeException: Unable to start HTTP server
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:432)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServer(VertxHttpRecorder.java:198)
	at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket-2064782366.deploy_0(VertxHttpProcessor$openSocket-2064782366.zig:119)
	at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket-2064782366.deploy(VertxHttpProcessor$openSocket-2064782366.zig:36)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:354)
	at io.quarkus.runtime.Application.start(Application.java:90)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:90)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:61)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
Caused by: java.util.concurrent.ExecutionException: java.net.BindException: Address already in use
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:403)
	... 10 more
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:461)
	at java.base/sun.nio.ch.Net.bind(Net.java:453)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
	at io.netty.channel.socket.nio.NioServerSocketChannel.doBind(NioServerSocketChannel.java:134)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:550)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1334)
	at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:506)
	at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:491)
	at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:973)
	at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:248)
	at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:356)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasksFrom(SingleThreadEventExecutor.java:428)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:377)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:488)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by famod** [comment]
Should be fixed in context of #9890 in 1.7.0.CR1.
